src/Windows/Avalonia.Win32/WindowImpl.cs

```diff
@@ -1440,15 +1460,19 @@ private void UpdateWindowProperties(WindowProperties newProperties, bool forceCh
                     style |= WindowStyles.WS_VISIBLE;
 
                 if (newProperties.IsResizable || newProperties.WindowState == WindowState.Maximized)
-                {
                     style |= WindowStyles.WS_THICKFRAME;
-                    style |= WindowStyles.WS_MAXIMIZEBOX;
-                }
                 else
-                {
                     style &= ~WindowStyles.WS_THICKFRAME;
+
+                if (newProperties.IsMinimizable)
+                    style |= WindowStyles.WS_MINIMIZEBOX;
+                else
+                    style &= ~WindowStyles.WS_MINIMIZEBOX;
+
+                if (newProperties.IsMaximizable || (newProperties.WindowState == WindowState.Maximized && newProperties.IsResizable))
+                    style |= WindowStyles.WS_MAXIMIZEBOX;
```

Comment:
I know it's annoying, but we need integration tests for win/mac :)

Reply:
I completely agree that we need more tests overall!
However, in this case, I don't really see a good way to test this with the current infrastructure.

Should we find the position of the maximize/minimize buttons and try to click them? (Can we even access the non-client area buttons via Appium?) Did you have another approach in mind?

Reply:
I wrote these extension methods before for window-decoration tests: https://github.com/AvaloniaUI/Avalonia/blob/master/tests/Avalonia.IntegrationTests.Appium/ElementExtensions.cs#L40

It might be enough to check enabled/visible state of these elements via appium apis.

Reply:
`GetSystemChromeButtons` for system titlebar. And `GetClientChromeButtons` for client titlebar (which also is properly recognized as an accessible titlebar, at least on windows).

Reply:
Thanks for the pointers! I really need to familiarize myself with the integration tests infrastructure.